src/components/ImageUploader/README.md

```diff
@@ -0,0 +1,225 @@
+		},
+
+		async uploadSuccessfulImage({ image, uploadProgressHandler }) {
+			return new Promise((resolve, reject) => {
+				this.uploadImage({
+					image,
+					uploadProgressHandler,
+					shouldSucceed: true,
+					resolve,
+					reject,
+				});
+			});
+		},
+
+		async uploadErrorImage({ image, uploadProgressHandler }) {
+			return new Promise((resolve, reject) => {
+				this.uploadImage({
+					image,
+					uploadProgressHandler,
+					shouldSucceed: false,
+					resolve,
+					reject,
+				});
+			});
+		},
+
+		uploadImage({
+			progress = 0,
+			resolve,
+			reject,
+			uploadProgressHandler,
+			image,
+			shouldSucceed,
+		}) {
+			let newProgress;
+
+			setTimeout(() => {
+				if (progress < 75) {
+					newProgress = progress + 15;
+					uploadProgressHandler(newProgress);
+					this.uploadImage({
+						resolve,
+						reject,
+						uploadProgressHandler,
+						progress: newProgress,
+						shouldSucceed,
+					});
+				} else if (shouldSucceed) {
+					uploadProgressHandler(100);
+					resolve({ data: { hello: 'world!', image } });
+				} else {
+					uploadProgressHandler(100);
+					reject(new Error('Some error'));
+				}
+			}, 500);
+		},
+	},
+};
+</script>
+
+<style module="$s">
+.SelectorContainer + .SelectorContainer {
+	margin-top: 32px;
+}
+
+.SelectorHeader {
+	margin-bottom: 16px;
+}
+</style>
+
+```
+
+<!-- api-tables:start -->
+## Props
+
+| Prop              | Type     | Default           | Possible values | Description                                                                                                                              |
+| ----------------- | -------- | ----------------- | --------------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| images            | `array`  | `[]`              | —               | List of images that have been selected                                                                                                   |
+| upload-handler-fn | `func`   | `() => undefined` | —               | Function called to trigger an upload. Called immediately on image selection, provided max size and max number image constraints are met. |
+| max-images        | `number` | `() => undefined` | —               | The maximum number of images allowed to be selected.                                                                                     |
+| max-size          | `number` | `() => undefined` | —               | The maximum file size allowed (in bytes)                                                                                                 |
+| accept            | `string` | `() => 'image/*'` | —               | Allowed file types, must be an image type (eg image/jpeg)                                                                                |
+
+
+## Events
+
+| Event | Type | Description |
+| ----- | ---- | ----------- |
+| input | -    | —           |
```

Comment:
When we have events with names that overlap standard HTML names, we normally prepend them with the component name to make it clear that it is a custom event. So in this case the event could be named `image-uplaoder:input`